src/app/login/page.tsx

```diff
@@ -0,0 +1,264 @@
+"use client";
+
+import { FormEvent, useState } from "react";
+import Link from "next/link";
+import { FaGoogle, FaFacebook } from "react-icons/fa";
+import { EyeIcon, EyeOffIcon } from "lucide-react";
+import { useRouter } from "next/navigation"; // Correct import for navigation in App Router
+
+import {
+  Card,
+  CardContent,
+  CardDescription,
+  CardHeader,
+  CardTitle,
+} from "@/components/ui/card";
+import { Input } from "@/components/ui/input";
+import { Button } from "@/components/ui/button";
+import { Label } from "@/components/ui/label";
+import { Separator } from "@/components/ui/separator";
+import { Checkbox } from "@/components/ui/checkbox";
+import { ModeToggle } from "@/components/ui/modeToggle";
+import { login } from "@/service/auth.service";
+import { useToast } from "@/hooks/use-toast";
+
+const LoginPage = () => {
+  const [showPassword, setShowPassword] = useState(false);
+  const [email, setEmail] = useState(""); // State for email
+  const [password, setPassword] = useState(""); // State for password
+  const [loading, setLoading] = useState(false); // State to manage loading
+  const [error, setError] = useState<string | null>(null); // State for general error handling
+  const [emailError, setEmailError] = useState(""); // State for email validation error
+  const [passwordError, setPasswordError] = useState(""); // State for password validation error
+  const router = useRouter(); // Use router from next/navigation for App Router
+
+  const { toast } = useToast();
+
+  // Email validation regex
+  const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
```

Comment:
Regex can be extracted to a constants file. That way it can be reused within the code base and the email validation remains consistent throughout the app.